find/main/decorators.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+
+from flask import abort, g
+
+from find.const import INTERNAL_DOMAIN
+
+
+def check_internal_user(func):
+    @wraps(func)
+    def decorated_function(*args, **kwargs):
+        internal_domains = tuple(INTERNAL_DOMAIN._value2member_map_.keys())
```

Comment:
I think we can avoid using internal attributes

```suggestion
        internal_domains = tuple(k.value for k in InternalDomain)
```

Reply:
Thanks @samuelhwilliams, just amended and pushed the last commit to include these suggested changes 👍